src/main/java/com/shop/ShopStorage.java

```diff
@@ -0,0 +1,5 @@
+package com.shop;
+
+public abstract class ShopStorage implements ShopOperations {
+
```

Comment:
I don't see the point of this abstract class. As a rule, abstract classes contain some default logic that will be used by inheritor classes.